README.rst

```diff
@@ -1,11 +1,138 @@
 django-pglock
-########################################################################
+#############
+
+``django-pglock`` performs advisory locks, table locks, and helps manage blocking locks.
+Here's some of the functionality at a glance:
+
+* ``pglock.advisory`` for application-level locking, for example, ensuring that tasks don't overlap.
+* ``pglock.model`` for locking an entire model.
+* ``pglock.timeout`` for dynamically setting the timeout to acquire a lock.
+* ``pglock.bypass`` to kill blocking locks for critical code, such as migrations.
+* The ``PGLock`` and ``BlockedPGLock`` models for querying active and blocked locks.
+* The ``pglock`` management command that wraps the models and provides other utilities.
+
+Quickstart
+==========
+
+Advisory Locks
+--------------
+
+Use ``pglock.advisory`` to acquire a `Postgres advisory lock <https://www.postgresql.org/docs/current/explicit-locking.html#ADVISORY-LOCKS>`__:
+
+.. code-block:: python
+
+    import pglock
+
+    with pglock.advisory("my_lock_id"):
+        # This code blocks until the "my_lock_id" lock is available
+
+
+Above our code will block until the lock is available, meaning
+no instances of the function will run simultaneously.
+
+Use ``wait=False`` or ``timeout=pglock.timedelta(seconds=...)`` to avoid waiting:
+
+.. code-block:: python
+
+    with pglock.advisory("my_lock_id", wait=False) as acquired:
+        if acquired:
+            # The lock is acquired
+
+Use ``side_effect=pglock.RAISE`` to raise a ``django.db.utils.OperationalError`` if
+the lock can't be acquired. When using the decorator, you can use
+``side_effect=pglock.SKIP`` to skip the function if the lock can't be acquired:
+
+.. code-block:: python
+
+    @pglock.advisory(wait=False, side_effect=pglock.SKIP)
+    def non_overlapping_func():
+        # This function will not run if there's another one already running.
+        # The decorator lock ID defaults to <module_name>.<function_name>
+
+Model Locks
+-----------
+
+``pglock.model`` can take a lock on an entire model during a transaction. For example:
+
+.. code-block:: python
+
+    from django.db import transaction
+    import pglock
+
+    with transaction.atomic():
+        pglock.model("auth.User")
```

Comment:
Maybe it's explained later, but I'm still not sure where one would actually use this in practice?